src/include/OpenImageIO/hash.h

```diff
@@ -81,6 +84,82 @@ inline size_t xxhash (const Str& s, size_t seed=1771) {
     return xxhash (s.data(), s.length(), seed);
 }
 
+template <typename T>
+class OIIO_API Hash {
```

Comment:
It would be great to have a comment here explaining the purpose and use of this class.